### PR TITLE
Allow full extensibility of this package by allowing custom blocks in an elegant way

### PR DESCRIPTION
**Pull request by TheDeadCode**
I saw #33 and I did not like the approach.

Instead, I've copied the semantics and how-to of the base Laravel Nova packages, and applied it to this package.

See Readme.md on how to extend this package.

I have re-implemented much of the functionality required to transform this package into a fully modular library. This makes it far easier to extend.

### CODE REVIEW

**Comment by TheDeadCode** [comment]
I have also made sure that this commit does not affect any current implementations.

File: Readme.md [hunk 1]
```diff
@@ -18,21 +18,131 @@ Publish the config file
 php artisan vendor:publish --provider="Advoor\NovaEditorJs\FieldServiceProvider"
 ```
 
+## Extending
```
Comment (roelofr) [inline Readme.md:21]: Makes more sense to put this after `Usage`, given that it's added behaviour.

File: Readme.md [hunk 2]
```diff
@@ -18,21 +18,131 @@ Publish the config file
...
+// ...
+```
+
+That's it for extending the Nova EditorJS package!
+
+
 ## Upgrade
 If upgrading from v0.4.0, re-publish the config file!
 
 ## Usage:
```
Comment (roelofr) [inline Readme.md:134]: While we're at it

```suggestion
## Usage
```

**Comment by roelofr** [comment]
I clicked approve a bit too enthusiasticly. Could you
[update the `let` to `const`][1] and [move the `Extending` section][2], then we can merge this.

[1]: https://github.com/advoor/nova-editor-js/pull/41#discussion_r463466740
[2]: https://github.com/advoor/nova-editor-js/pull/41#discussion_r463466899

File: resources/js/components/FormField.vue [hunk 3]
```diff
@@ -186,32 +24,13 @@
...
-                setInlineCodeToolSettings(self, tools);
-                setChecklistToolSettings(self, tools);
-                setMarkerToolSettings(self, tools);
-                setDelimiterToolSettings(self, tools);
-                setTableToolSettings(self, tools);
-                setRawToolSettings(self, tools);
-                setEmbedToolSettings(self, tools);
 
-                var editor = new EditorJS({
+                let editor = NovaEditorJS.getInstance({
```
[the remark this answers is not in the document]
Comment (TheDeadCode) [reply resources/js/components/FormField.vue]: Done

File: Readme.md (hunk 1 above)
Comment (TheDeadCode) [reply Readme.md:66]: Done

File: resources/js/nova-editor.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+    getInstance(config, field) {
+
+        let editorConfig = _.merge({}, this.defaultConfigObject, config);
+        const fieldObject = _.cloneDeep(field);
+
+        // Plugins should not modify the field config.
+        // If a key should be changed, other plugins loaded later
+        // would have an unsynchronized version of the field configuration.
+        Object.freeze(fieldObject);
```
Comment (TheDeadCode) [inline resources/js/nova-editor.js:44]: It's also good ot know I am freezing a deep clone of the field object, to ensure plugins do not change it.

File: src/config/nova-editor-js.php [hunk 5]
```diff
@@ -249,6 +249,7 @@
                             'data' => [
                                 '-' => [
                                     'type' => 'string',
+                                    'allowedTags' => 'i,b,u,a[href],span[class],code[class],mark[class]'
```
Comment (TheDeadCode) [inline src/config/nova-editor-js.php:252]: I have also changed added allowed tags, to avoid any confusion on why the tables do not show any formatting.

Comment (roelofr) [reply src/config/nova-editor-js.php:252]: I don't think this should be an issue, but I'm gonna mark it as breaking anyway (since the impact is unknown)

**Pull request merged**